reactive/swarm.py

```diff
@@ -25,9 +26,17 @@ def swarm_etcd_cluster_setup(etcd):
     """
     bind_docker_daemon()
     con_string = etcd.connection_string().replace('http', 'etcd')
-    start_swarm_etcd_agent(con_string)
-    if hookenv.is_leader():
-        start_swarm_etcd_manager(con_string)
+
+    opts = {}
+    opts['connection_string'] = con_string
+    opts['addr'] = hookenv.unit_private_ip()
+    opts['port'] = 2375
+    opts['leader'] = is_leader()
```

Comment:
It will, so long as the swarm.available state is removed. That will trigger the unit to cycle. Not having a juju version to test this on causes a problem but once thats resolved we can test the theory and make an assertion